containers/src/Data/IntSet/Internal.hs

```diff
@@ -334,7 +363,7 @@ null _   = False
 size :: IntSet -> Int
 size = go 0
   where
-    go !acc (Bin _ _ l r) = go (go acc l) r
+    go !acc (Bin _ l r) = go (go acc l) r
     go acc (Tip _ bm) = acc + bitcount 0 bm
     go acc Nil = acc
```

Comment:
Why does `bitcount` take two arguments? Should we just use `popCount` directly?

Reply:
> Why does `bitcount` take two arguments?

This seems to be left over from when `bitcount` was a loop and the first parameter was the accumulator: https://github.com/haskell/containers/commit/4952822250db26c2c6d586abef3f8745b2c6e3a8#diff-06e022e2fe36764ea9baec24a03a8186d708cc561677a2e6583f08fe180ca073L73-L76

What I don't understand is why it is `acc + bitcount 0 bm` rather than `bitcount acc bm` 😄 

> Should we just use `popCount` directly?

Probably. Might be better in a different PR though.

Reply:
I think it would be a totally reasonable "by the way" for this one, if you're so inclined. It strictly improves readability.

Reply:
Also, we should get rid of the note citing a source for the highest bit algorithm that we no longer use.

Reply:
Since we were considering removing the `Nat` stuff in #995, I think it would a better fit with those changes. Same for the obsolete note.